pkg/org/organizations_test.go

```diff
@@ -0,0 +1,287 @@
+	}
+}
+
+func TestReadVersionAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	vJSON, err := service.ReadVersionAsJSON(ctx, id2, id2)
+	if expect.NoError(err) {
+		expect.Contains(string(vJSON), id2)
+	}
+}
+
+func TestReadAllVersions(t *testing.T) {
+	expect := assert.New(t)
+	allVersions, err := service.ReadAllVersions(ctx, id2)
+	if expect.NoError(err) {
+		expect.Equal(o20, allVersions[0])
+	}
+}
+
+func TestReadAllVersionsAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	versionsCheckJSON, err := service.ReadAllVersionsAsJSON(ctx, id3)
+	if expect.NoError(err) {
+		expect.Contains(string(versionsCheckJSON), id3)
+	}
+}
+
+// I don't fully believe in a reliability of this test
+// talk to Dave about it on a next meeting
+func TestReadNames(t *testing.T) {
+	expect := assert.New(t)
+	idsAndNames, err := service.ReadNames(ctx, false, 3, "")
+	onlyNames := make([]string, 0, len(idsAndNames))
+
+	for _, kv := range idsAndNames {
+		onlyNames = append(onlyNames, kv.Value)
+	}
+	if expect.NoError(err) {
+		expect.Equal(onlyNames, knownNames)
+	}
+}
+
+func TestReadAllNames(t *testing.T) {
+	expect := assert.New(t)
+	idsAndNames, err := service.ReadAllNames(ctx, true)
+
+	expectedNames := []v.TextValue{
+		{Value: o11.Name},
+		{Value: o20.Name},
+		{Value: o30.Name},
+	}
+
+	if expect.NoError(err) {
+		reflect.DeepEqual(idsAndNames, expectedNames)
+	}
+}
+
+func TestFilterNames(t *testing.T) {
+	expect := assert.New(t)
+	filteredName, err := service.FilterNames(ctx, "1.1", false)
+
+	expectedName := []v.TextValue{
+		{
+			Key:   o11.ID,
+			Value: o11.Name,
+		},
+	}
+
+	if expect.NoError(err) {
+		expect.Equal(filteredName, expectedName)
+	}
+}
+
+func TestReadStatuses(t *testing.T) {
+	expect := assert.New(t)
+	// Read all statuses
+	allStatuses := []string{"PENDING", "ENABLED", "DISABLED"}
+	statuses, err := service.ReadStatuses(ctx, false, 3, tuid.MinID)
+	if expect.NoError(err) {
+		expect.GreaterOrEqual(len(statuses), len(allStatuses))
+		expect.Subset(statuses, allStatuses)
+	}
+}
+
+func TestReadAllStatuses(t *testing.T) {
+	expect := assert.New(t)
+	// Read all statuses
+	allStatuses := []string{"PENDING", "ENABLED", "DISABLED"}
+	statuses, err := service.ReadAllStatuses(ctx)
+	if expect.NoError(err) {
+		expect.Subset(statuses, allStatuses)
+	}
+}
+
+func TestReadOrganizationsByStatus(t *testing.T) {
+	expect := assert.New(t)
+	checkOrgs, err := service.ReadOrganizationsByStatus(ctx, "DISABLED", false, 1, "")
+	if expect.NoError(err) {
+		expect.Equal(o30, checkOrgs[0])
+	}
+}
+
+func TestReadOrganizationByStatusAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	checkOrgs, err := service.ReadOrganizationsByStatusAsJSON(ctx, "PENDING", false, 1, "")
+	if expect.NoError(err) {
+		expect.Contains(string(checkOrgs), o10.Status)
+	}
+}
+
+func TestReadAllOrganizationsByStatus(t *testing.T) {
+	expect := assert.New(t)
+	checkOrgs, err := service.ReadAllOrganizationsByStatus(ctx, "ENABLED")
+	if expect.NoError(err) {
```

Comment:
We could make this test a little more robust by checking the number of organizations returned. There should be at least 2 (organizations 1 and 2), and maybe 3 (organization 4). We could also check for the subset of organization IDs 1 and 2.